### PR TITLE
add GetUsersByGroupID

### PR DESCRIPTION
**Pull request by paoloromolini**
Adding get users by group id function.

### CODE REVIEW

**Comment by JinHuangAtZen** [comment]
Hi @paoloromolini , please refer to this doc https://github.com/nukosuke/go-zendesk/blob/master/CBPMigration.md, Zendesk is obsoleting OBP (Offset Based Pagination), please follow the new way to create CBP (Cursor Based Pagination) iterators and starting to adopt CBP in your code base. 
Thank you

**Comment by paoloromolini** [comment]
Hi @JinHuangAtZen I've added a new commit following your instructions. Please let me know if it is correct.

File: zendesk/user.go [hunk 1]
```diff
@@ -128,6 +129,8 @@ type UserAPI interface {
 	GetOrganizationUsersIterator(ctx context.Context, opts *PaginationOptions) *Iterator[User]
 	GetOrganizationUsersOBP(ctx context.Context, opts *OBPOptions) ([]User, Page, error)
 	GetOrganizationUsersCBP(ctx context.Context, opts *CBPOptions) ([]User, CursorPaginationMeta, error)
+	GetGroupUsersOBP(ctx context.Context, opts *OBPOptions) ([]User, Page, error)
```
Comment (JinHuangAtZen) [inline zendesk/user.go:132]: This is missing the generated `GetGroupUsersIterator(ctx context.Context, opts *PaginationOptions) *Iterator[User]`

File: zendesk/user.go [hunk 2]
```diff
@@ -116,6 +116,7 @@ type UserAPI interface {
 	SearchUsers(ctx context.Context, opts *SearchUsersOptions) ([]User, Page, error)
 	GetManyUsers(ctx context.Context, opts *GetManyUsersOptions) ([]User, Page, error)
 	GetUsers(ctx context.Context, opts *UserListOptions) ([]User, Page, error)
+	GetUsersByGroupID(ctx context.Context, groupID int64, opts *UserListOptions) ([]User, Page, error)
```
Comment (JinHuangAtZen) [inline zendesk/user.go:119]: This is still using the old OBP `UserListOptions` and this function shouldn't be exist, it should be replaced by the newly generated `GetGroupUsersIterator` function.

File: zendesk/user.go [hunk 3]
```diff
@@ -256,7 +259,35 @@ func (z *Client) GetManyUsers(ctx context.Context, opts *GetManyUsersOptions) ([
 	return data.Users, data.Page, nil
 }
 
-//TODO: GetUsersByGroupID, GetUsersByOrganizationID
+// GetUsersByGroupID get users by group ID
+// ref: https://developer.zendesk.com/api-reference/ticketing/users/users/#list-users
+func (z *Client) GetUsersByGroupID(ctx context.Context, groupID int64, opts *UserListOptions) ([]User, Page, error) {
```
Comment (JinHuangAtZen) [inline zendesk/user.go:264]: This is still using the old OBP UserListOptions and this function shouldn't be exist, it should be replaced by the newly generated GetGroupUsersIterator function.

**Comment by JinHuangAtZen** [comment]
> Hi @JinHuangAtZen I've added a new commit following your instructions. Please let me know if it is correct.

Hi @paoloromolini , Thanks for adding the new methods, I've added some comments, basically, the new `GetGroupUsersIterator` will replace the `GetUsersByGroupID`

**Pull request closed without merging**